common/model-views.cpp

```diff
@@ -2195,7 +2195,74 @@ namespace rs2
                 static_cast<float>(vd.height()) };
         }
         _stream_not_alive.reset();
+        
+        try 
+        {
+            auto ds = d->dev.first<depth_sensor>();
+            if( ds.supports( RS2_OPTION_ENABLE_IR_REFLECTIVITY )
+                    && ds.supports( RS2_OPTION_ENABLE_MAX_USABLE_RANGE )
+                    && ( p.stream_type() == RS2_STREAM_INFRARED ) || ( p.stream_type() == RS2_STREAM_DEPTH ) )
+            {
+                _reflectivity = std::unique_ptr< reflectivity >(new reflectivity());
```

Comment:
Can't we check whether it's skipped here rather than later?

Reply:
👍